Panic when merging sugar into config with invalid file reference

Input:
```
variant: fcos
version: 1.3.0
ignition:
  config:
    merge:
      - local: missing-file
boot_device:
  mirror:
    devices:
      - /dev/sda
      - /dev/sdb
```

Output:
```
panic: mismatched translation tags, yaml !=  || json != 

goroutine 1 [running]:
github.com/coreos/fcct/base/util.MergeTranslatedConfigs(0x766900, 0xc0001684e0, 0x790d02, 0x4, 0x790c6e, 0x4, 0xc00016d230, 0x766900, 0xc000168680, 0x0, ...)
	/home/bgilbert/coreos/fcct/base/util/merge.go:38 +0x739
github.com/coreos/fcct/config/fcos/v1_3.Config.processBootDevice(0xc000124208, 0x5, 0xc0001241f0, 0x4, 0xc000126700, 0x1, 0x1, 0x0, 0x0, 0x0, ...)
	/home/bgilbert/coreos/fcct/config/fcos/v1_3/translate.go:248 +0x2315
github.com/coreos/fcct/config/fcos/v1_3.Config.ToIgn3_2Unvalidated(0xc000124208, 0x5, 0xc0001241f0, 0x4, 0xc000126700, 0x1, 0x1, 0x0, 0x0, 0x0, ...)
	/home/bgilbert/coreos/fcct/config/fcos/v1_3/translate.go:66 +0x2a7
reflect.Value.call(0x76f960, 0xc00013e900, 0x693, 0x790b06, 0x4, 0xc0000e0de0, 0x1, 0x1, 0x40ad5f, 0x74f1c0, ...)
	/usr/lib/golang/src/reflect/value.go:476 +0x8c7
reflect.Value.Call(0x76f960, 0xc00013e900, 0x693, 0xc0000e0de0, 0x1, 0x1, 0x0, 0x0, 0x722720)
	/usr/lib/golang/src/reflect/value.go:337 +0xb9
github.com/coreos/fcct/config/util.Translate(0x76f960, 0xc00013e900, 0x7953fa, 0x13, 0x7ffcb4f4f948, 0x1, 0x790d00, 0x4, 0xc000126d80, 0x3, ...)
	/home/bgilbert/coreos/fcct/config/util/util.go:59 +0x34e
github.com/coreos/fcct/config/fcos/v1_3.Config.ToIgn3_2(...)
	/home/bgilbert/coreos/fcct/config/fcos/v1_3/translate.go:75
reflect.Value.call(0x754b00, 0xc00013e240, 0x213, 0x790b06, 0x4, 0xc0000e1c20, 0x1, 0x1, 0x40ad5f, 0x74f1c0, ...)
	/usr/lib/golang/src/reflect/value.go:476 +0x8c7
reflect.Value.Call(0x754b00, 0xc00013e240, 0x213, 0xc0000e1c20, 0x1, 0x1, 0xc00013e240, 0x213, 0x0)
	/usr/lib/golang/src/reflect/value.go:337 +0xb9
github.com/coreos/fcct/config/util.TranslateBytes(0xc000134000, 0x99, 0x600, 0x754b00, 0xc00013e240, 0x791b49, 0x8, 0x7ffcb4f4f948, 0x1, 0xc000124100, ...)
	/home/bgilbert/coreos/fcct/config/util/util.go:110 +0x39c
github.com/coreos/fcct/config/fcos/v1_3.ToIgn3_2Bytes(0xc000134000, 0x99, 0x600, 0x7ffcb4f4f948, 0x1, 0x0, 0x0, 0x0, 0x0, 0x7a8e50, ...)
	/home/bgilbert/coreos/fcct/config/fcos/v1_3/translate.go:83 +0xd3
github.com/coreos/fcct/config.TranslateBytes(0xc000134000, 0x99, 0x600, 0x7ffcb4f4f948, 0x1, 0x600, 0x0, 0x0, 0x2f, 0x9b51c0, ...)
	/home/bgilbert/coreos/fcct/config/config.go:102 +0x2ce
main.main()
	/home/bgilbert/coreos/fcct/internal/main.go:93 +0x5df
```

--- a/config/fcos/v1_3/translate.go
+++ b/config/fcos/v1_3/translate.go
@@ -63,6 +63,9 @@ const (
 // validation is performed on input or output.
 func (c Config) ToIgn3_2Unvalidated(options common.TranslateOptions) (types.Config, translate.TranslationSet, report.Report) {
 	ret, ts, r := c.Config.ToIgn3_2Unvalidated(options)
+	if r.IsFatal() {
+		return types.Config{}, translate.TranslationSet{}, r
+	}
 	r.Merge(c.processBootDevice(&ret, &ts, options))
 	return ret, ts, r
 }

--- a/config/fcos/v1_4_exp/translate.go
+++ b/config/fcos/v1_4_exp/translate.go
@@ -63,6 +63,9 @@ const (
 // validation is performed on input or output.
 func (c Config) ToIgn3_3Unvalidated(options common.TranslateOptions) (types.Config, translate.TranslationSet, report.Report) {
 	ret, ts, r := c.Config.ToIgn3_3Unvalidated(options)
+	if r.IsFatal() {
+		return types.Config{}, translate.TranslationSet{}, r
+	}
 	r.Merge(c.processBootDevice(&ret, &ts, options))
 	return ret, ts, r
 }

